packages/nodejs/src/express/abbyMiddlewareFactory.ts

```diff
@@ -0,0 +1,89 @@
+import { NextFunction, Request, Response } from "express";
+import { setRequest } from "../abby/contexts/requestContext";
+import { setResponse } from "../abby/contexts/responseContext";
+import { AbbyConfig, ABConfig, FlagValueString } from "@tryabby/core";
+import { createAbby } from "../abby/createAbby";
+import { F } from "ts-toolbelt";
+
+interface AbbyMiddlewareFactoryResult {
+  featureFlagMiddleware: <F extends string>(
+    name: F,
+    req: Request,
+    res: Response,
+    next: NextFunction,
+    deciderFunction: (req: Request, flagValue: any) => boolean
+  ) => void;
+  allTestsMiddleWare: <T extends string>(req: Request, res: Response, next: NextFunction) => void;
+  getVariant: <T extends string>(name: T) => any;
+}
+
+export const abbyMiddlewareFactory = <
+  FlagName extends string,
+  TestName extends string,
+  Tests extends Record<TestName, ABConfig>,
+  Flags extends Record<FlagName, FlagValueString> = Record<FlagName, FlagValueString>,
+  ConfigType extends AbbyConfig<FlagName, Tests> = AbbyConfig<FlagName, Tests>,
+>({
+  abbyConfig,
+}: {
+  abbyConfig: F.Narrow<AbbyConfig<FlagName, Tests>>;
+}) => {
+  const setCookie = (cookie: string) => {
+    throw new Error("Method not implemented.");
```

Comment:
the message can be better. have a clear signature so you know where to go. `abbyMiddlewareFactory.setCookie(): your message` for example